core/src/test/java/org/elasticsearch/indices/IndicesServiceTests.java

```diff
@@ -81,6 +81,8 @@ public void testCanDeleteShardContent() {
         assertFalse("shard is allocated", indicesService.canDeleteShardContent(shardId, test.getIndexSettings()));
         test.removeShard(0, "boom");
         assertTrue("shard is removed", indicesService.canDeleteShardContent(shardId, test.getIndexSettings()));
+        ShardId notAllocated = new ShardId(test.index(), 100);
```

Comment:
we test that shard 100 can not be deleted because it's not there *when* shard 0 is removed. We don't test when shard 0 is present .